Description: correctly parse kcmdline parameters containing a dot
 Kernel command line parameters of modules were not parsed correctly if
 the value contained a dot.
Origin: upstream, commit:66f3228d17d66d7e2dd484427259290fbc82b2f0
Applied-Upstream: commit:66f3228d17d66d7e2dd484427259290fbc82b2f0
Bug-Debian: http://bugs.debian.org/689872
---

commit 66f3228d17d66d7e2dd484427259290fbc82b2f0
Author: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date:   Mon Oct 8 19:04:16 2012 -0300

    libkmod: Add support for '.' in module parameter on kcmdline
    
    Otherwise we fail to parse arguments in kernel command line like
    testmodule.testparam=1.5G
    
    Suggested-by: Selim T. Erdogan <selim@alumni.cs.utexas.edu>

diff --git a/libkmod/libkmod-config.c b/libkmod/libkmod-config.c
index 70044f0..398468e 100644
--- a/libkmod/libkmod-config.c
+++ b/libkmod/libkmod-config.c
@@ -567,8 +567,10 @@ static int kmod_config_parse_kcmdline(struct kmod_config *config)
 			modname = p + 1;
 			break;
 		case '.':
-			*p = '\0';
-			param = p + 1;
+			if (param == NULL) {
+				*p = '\0';
+				param = p + 1;
+			}
 			break;
 		case '=':
 			if (param != NULL)
